lib/facter/cisco.rb

```diff
@@ -8,48 +9,6 @@
   end
 
   setcode do
-    hash = {}
-
-    hash['images'] = {}
-    begin
-      hash['images']['system_image'] = Platform.system_image
-      hash['images']['full_version'] = Platform.image_version
-    rescue NameError
-      # In more recent versions, Platform moved into the Cisco namespace.
-      Platform = Cisco::Platform
-      Feature = Cisco::Feature
-      hash['images']['system_image'] = Platform.system_image
-      hash['images']['full_version'] = Platform.image_version
-    end
-    hash['images']['packages'] = Platform.packages
-
-    hash['hardware'] = {}
-    hash['hardware']['type'] = Platform.hardware_type
-    hash['hardware']['cpu'] = Platform.cpu
-    hash['hardware']['memory'] = Platform.memory
-    hash['hardware']['board'] = Platform.board
-    hash['hardware']['last_reset'] = Platform.last_reset
-    hash['hardware']['reset_reason'] = Platform.reset_reason
-
-    hash['inventory'] = {}
-    hash['inventory']['chassis'] = Platform.chassis
-    Platform.slots.each do |slot, info|
-      hash['inventory'][slot] = info
-    end
-    Platform.power_supplies.each do |ps, info|
-      hash['inventory'][ps] = info
-    end
-    Platform.fans.each do |fan, info|
-      hash['inventory'][fan] = info
-    end
-
-    hash['virtual_service'] = Platform.virtual_services
-
-    hash['feature_compatible_module_iflist'] = {}
-    interface_list = Feature.compatible_interfaces('fabricpath')
-    hash['feature_compatible_module_iflist']['fabricpath'] = interface_list
-    hash['hardware']['uptime'] = Platform.uptime
-
-    hash
+    Facter::CiscoNexus.platform_facts
```

Comment:
What's the rational behind moving this into a separate `CiscoNexus` namespace?  Future support of other Cisco OS's?

Reply:
Namespaced to match the device type, and in case there are future OS's.